source/val/validate_conversion.cpp

```diff
@@ -273,6 +273,18 @@ spv_result_t ConversionPass(ValidationState_t& _, const Instruction* inst) {
           return _.diag(SPV_ERROR_INVALID_DATA, inst)
                  << "Pointer storage class must be PhysicalStorageBufferEXT: "
                  << spvOpcodeString(opcode);
+
+        if (spvIsVulkanEnv(_.context()->target_env)) {
```

Comment:
I think you have these checks reversed. This is checking OpConvertPtrToU and should be checking result type.

Reply:
ok, so the current VU says

> OpConvertUToPtr and OpConvertPtrToU must use an integer type whose Width is 64

And I realized I read the VU as the `OpTypePointer` type must be pointing to a 64 wide int

Ok, this makes a LOT more sense, will reverse the two and clear up the error message